RFC-005-SWAP-Bitcoin-basic.md

```diff
@@ -0,0 +1,248 @@
+| redeem_signature | A valid SECP256k1 ECDSA DER encoded signature on the transaction with respect to the `redeem_pubkey`    |
+| redeem_pubkey    | The 33 byte SECP256k1 compressed public key that was hashed to produce the pubkeyhash `redeem_identity` |
+| secret           | The pre-image of the `secret_hash` under the `hash_function`                                            |
+| `01`             | A single byte used to activate the redeem path in the `OP_IF`                                           |
+| contract_script  | The compiled contract (as generally required when redeeming from a P2WSH output)                        |
+
+For how to use this witness data to construct the redeem transaction see [BIP141](https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki#transaction-id).
+
+To be notified of the redeem event, both parties MAY watch the blockchain for transactions that spend from the output and check that the witness data is in the above form.
+If Bitcoin is the `beta_ledger`, then the funder (Bob) MUST watch for such a transaction and then extract the `secret` from the witness data and continue the protocol.
+
+### Refund
+
+To refund the HTLC, the funder MUST submit a transaction to the blockchain which spends the P2WSH output.
+The funder can use the following witness data to spend the output after the `expiry`:
+
+| Data             | Description                                                                                             |
+|:-----------------|:--------------------------------------------------------------------------------------------------------|
+| refund_signature | A valid SECP256k1 ECDSA DER encoded signature on the transaction with respect to the `refund_pubkey`    |
+| refund_pubkey    | The 33 byte SECP256k1 compressed public key that was hashed to produce the pubkeyhash `refund_identity` |
+| `00`             | A single byte used to activate the refund path in the `OP_IF`                                           |
+| contract_script  | The compiled contract (as generally required when redeeming from a P2WSH output)                        |
+
+To be notified of the refund event, both parties MAY watch the blockchain for transactions that spend from the output and check that the witness data is in the above form.
+
+## Registry extension
+
+This RFC extends the [registry](./registry.md#identities) with an identity definition for the Bitcoin ledger:
+
+| Ledger  | Identity Name | JSON Encoding            | Description                                                                           |
+|:--------|:--------------|:-------------------------|---------------------------------------------------------------------------------------|
+| Bitcoin | `pubkeyhash`  | `hex-encoded-bytes (20)` | The result of applying SHA-256 and then RIPEMD-160 to a SECP256k1 compressed public key |
+
+# Examples
```

Comment:
:wrench: Could be explicit that it contains test vectors.

```suggestion
# Examples and test vectors
```

Reply:
Changed to Examples/Test vectors